src/main/lrsql/util.clj

```diff
@@ -25,15 +25,19 @@
    folloiwing error data is thrown:
      :type       ::parse-failure
      :data       `data`
-     :data-type  `data-type`"
-  [parse-fn data-type data]
+     :data-type  `data-type`
+   You may add a keyword arg :retry-parse-fn if you would like a backup function
+   to be attempted instead on parse failure."
+  [parse-fn data-type data & {:keys [retry-parse-fn]}]
   `(try (~parse-fn ~data)
         (catch Exception e#
-          (throw (ex-info (format "Cannot parse nil or invalid %s"
-                                  ~data-type)
-                          {:type      ::parse-failure
-                           :data      ~data
-                           :data-type ~data-type})))))
+          ~(if (some? retry-parse-fn)
+             `(wrap-parse-fn ~retry-parse-fn ~data-type ~data)
+             `(throw (ex-info (format "Cannot parse nil or invalid %s"
+                                      ~data-type)
+                              {:type      ::parse-failure
+                               :data      ~data
+                               :data-type ~data-type}))))))
```

Comment:
Ahh, this looks familiar

Reply:
Who's to say from where inspiration springs.